Datadog Metric Slicing

Jtang-1 opened the pull request:
## Product Description
<!-- For non-invisible changes, describe user-facing effects. -->

This related Formplayer PR affects metric `formplayer.metrics.timings`.
- Datadog tag `module_initiated_by` is renamed to `request_initiated_by`.

This PR adds the values `pagination` and `clickable_icon` to capture requests that are triggered by user action pagination and clickable icon.

![Screen Shot 2024-03-22 at 9 56 21 PM](https://github.com/dimagi/commcare-hq/assets/88759246/ba6a5347-7d1b-47a5-8c21-6bfa2cf02e46)



## Technical Summary
<!--
    Provide a link to the ticket or document which prompted this change,
    Describe the rationale and design decisions.
-->
[Jira Ticket](https://dimagi.atlassian.net/browse/USH-4182)
[Related Formplayer Ticket](https://github.com/dimagi/formplayer/pull/1551)

See FP PR for reasoning of changing `module_initiated_by` tag.

## Feature Flag
<!-- If this is specific to a feature flag, which one? -->
no FF

## Safety Assurance

### Safety story
<!--
Describe how you became confident in this change, such as
local testing, why the change is inherently safe, and/or plans to limit the blast radius of a defect.

In particular consider how existing data may be impacted by this change.
-->
Will test on staging. Blast radius is contained to clickable icons, pagination, dynamic search, and case search validation.

### Automated test coverage

<!-- Identify the related test coverage and the tests it would catch -->
no automated tests

### QA Plan

<!--
- Describe QA plan that along with automated test coverages proves this PR is regression free
- Link to QA Ticket
-->
QA to ensure no impact on clickable icons, pagination, dynamic search, and case search validation.
 [QA Ticket](https://dimagi.atlassian.net/browse/QA-6188?focusedCommentId=316724&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-316724)
### Rollback instructions

<!--
If this PR follows standards of revertability, check the box below.
Otherwise replace it with detailed instructions or reasons a rollback is impossible.
-->

- [X] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [X] Risk label is set correctly
- [X] The set of people pinged as reviewers is appropriate for the level of risk of the change

Jtang-1 committed:
prefactor: extract initiatedBy tag out of queryData

Jtang-1 committed:
prefactor: rename constant and account for requestInitiatedByTags extracted from queryData

Jtang-1 committed:
feat: adds tag for pagination and clickable icon user action

Jtang-1 committed:
add comments

MartinRiese reviewed (approved):
Looks good except for the lint error

Jtang-1 committed:
bug: query input needs to be included in queryData when converting from json

Jtang-1 committed:
remove duplicated import

Jtang-1 commented on corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/views.js:
```diff
@@ -9,7 +9,8 @@ hqDefine("cloudcare/js/formplayer/menus/views", function () {
         cloudcareUtils = hqImport("cloudcare/js/utils"),
-        markdown = hqImport("cloudcare/js/markdown");
+        markdown = hqImport("cloudcare/js/markdown"),
+        formplayerConstants = hqImport("cloudcare/js/formplayer/constants");
```
ah yes thank you [1e9e6e7](https://github.com/dimagi/commcare-hq/pull/34247/commits/1e9e6e7dac7d478c64949284b35b7714d8d4ef12)

MartinRiese committed:
Merge branch 'master' into jt/datadog-metric-slicing

Jtang-1 committed:
Merge remote-tracking branch 'origin' into jt/datadog-metric-slicing

Jtang-1 committed:
requestInitiatedBy should be associated with only one value

There shouldn't be multiple values associated with one request for the datadog
request_initiated_by tag
